src/coreclr/jit/indirectcalltransformer.cpp

```diff
@@ -273,21 +273,32 @@ class IndirectCallTransformer
             {
                 assert(currBlock->KindIs(BBJ_ALWAYS));
                 currBlock->SetTarget(checkBlock);
-                compiler->fgAddRefPred(checkBlock, currBlock);
+                FlowEdge* const newEdge = compiler->fgAddRefPred(checkBlock, currBlock);
+                newEdge->setLikelihood(1.0);
             }
 
             // checkBlock
+            // Todo: get likelihoods right
```

Comment:
It's not in scope for this PR, but I'm curious what your ideas are for coming up with better guesses for the likelihoods in cases like this one.

Reply:
For that specific case the logic is in a base class so we will need the various derived classes to make the right info available.

Generally speaking if you are introducing new conditional flow you should have some idea of the relative frequencies of the different possibilities. If not you can always go off and try and do some custom measurements to see what happens.

One example where I may do that is in the "OSR step block" case -- here I do not know the right answer, my guess would be that it depends on whether the try that encompasses the OSR entry point is itself nested in a loop. But to be sure I'd have to go measure.